src/Sylius/Bundle/ApiBundle/EventSubscriber/TaxonDeletionEventSubscriber.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\ApiBundle\EventSubscriber;
+
+use ApiPlatform\Core\EventListener\EventPriorities;
+use Sylius\Component\Core\Model\TaxonInterface;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpKernel\Event\ViewEvent;
+use Symfony\Component\HttpKernel\KernelEvents;
+use Symfony\Component\Messenger\MessageBusInterface;
+use Sylius\Component\Channel\Repository\ChannelRepositoryInterface;
+use Symfony\Component\HttpKernel\Exception\BadRequestHttpException;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Sylius\Bundle\ApiBundle\Exception\CannotRemoveMenuTaxonException;
+
+final class TaxonDeletionEventSubscriber implements EventSubscriberInterface
```

Comment:
```suggestion
/** @experimental */
final class TaxonDeletionEventSubscriber implements EventSubscriberInterface
```

we mark all in ApiBundle as experimental as this bundle is under development.
And as @Zales0123 notices, the event subscriber needs to be tested with phpspec.